### PR TITLE
Filter dropdown and typeahead loading states

### PR DESCRIPTION
**Pull request by xtine**
Supports loading interactions for filter dropdowns and typeahead inputs.

Supports:
18F/fec-style#451 (filter dropdown interactions)
18F/fec-style#452 (filter typeahead loading states)

### CODE REVIEW

File: static/js/modules/tables.js [hunk 1]
```diff
@@ -320,10 +309,17 @@ function filterSuccessUpdates(changeCount) {
       clearTimeout(messageTimer);
     }
 
-    $label.after($('<div class="filter__message filter__message--success">' + message + '</div>')
-      .hide().fadeIn());
+    // Clicking on "Clear all filters" will remove all dropdown checkboxes,
+    // so we check to make sure the message isn't shown inside the dropdown panel.
+    if (!$label.parent().parent().hasClass('dropdown__list')) {
```
Comment (noahmanger) [inline static/js/modules/tables.js]: What about using `$label.closest('.dropdown__list').length < 1` to be a little more concise?

File: static/js/modules/tables.js [hunk 2]
```diff
@@ -259,17 +259,10 @@ function filterSuccessUpdates(changeCount) {
     var filterAction = '';
     var $filterMessage = $('.filter__message');
 
+    $('.is-successful').removeClass('is-successful');
+
     if (type === 'checkbox' || type === 'radio') {
       $label = $('label[for="' + updateChangedEl.id + '"]');
-      $('.is-successful').removeClass();
```
Comment (noahmanger) [inline static/js/modules/tables.js]: Where did all this stuff that was removed end up?

Comment (noahmanger) [inline static/js/modules/tables.js]: Where did all this stuff that was removed end up?

Comment (xtine) [inline static/js/modules/tables.js]: I just combined logic together and it seems more straightforward.

**Pull request merged**